python/GafferUI/NodeMenu.py

```diff
@@ -116,7 +116,8 @@ def f( menu ) :
 				if node is None :
 					return
 
-				Gaffer.NodeAlgo.applyUserDefaults( node )
+				with script.context():
+					Gaffer.NodeAlgo.applyUserDefaults( node )
```

Comment:
I don't think this is necessarily a bad idea (we do the same when evaluating presets), but there are other places where we apply user defaults as well - I haven't looked to see if the same is possible there. Maybe it's OK to start with just this one place?

I suppose the other alternative would be to call `applyUserDefaults()` after parenting the node, and have `applyUserDefaults()` come up with the context internally?